src/components/service-room.ts

```diff
@@ -0,0 +1,167 @@
+import { MatrixClient } from "matrix-bot-sdk";
+
+export enum ServiceNotificationNoticeCode {
+	Unknown = "UNKNOWN",
+	Blocked = "BLOCKED",
+	RemoteServiceOutage = "REMOTE_SERVICE_OUTAGE",
+	MatrixServiceOutage = "MATRIX_SERVICE_OUTAGE",
+}
+
+export enum ServiceNotificationServerity {
+	/**
+	 * Just information for the administrator, usually no action required.
+	 */
+	Infomational = "info",
+	/**
+	 * Something the administrator should know about, might require an explicit notification.
+	 */
+	Warning = "warning",
+	/**
+	 * A serious issue has occured with the bridge, action needed.
+	 */
+	Error = "error",
+	/**
+	 * The bridge cannot function, action urgently needed.
+	 */
+	Critical = "critical"
+}
+
+
+export interface ServiceRoomOpts {
+	/**
+	 * The roomId to send notices to.
+	 */
+	roomId: string;
+	/**
+	 * The minimum time allowed before
+	 * a new notice with the same ID can be sent (to avoid room spam).
+	 * Defaults to a hour.
+	 */
+	minimumUpdatePeriodMs?: number;
```

Comment:
How about making this non-optional and setting it to a default value in the constructor?

This would also ensure that callers won't have to handle falling back to a default value when checking this.